### PR TITLE
[ios] Add custom events library start-up delay

### PR DESCRIPTION
**Pull request by rclee**
Adds a custom delay to the events library initialization. This is only active when using a certain profile configuration.

### CODE REVIEW

**Comment by rclee** [comment]
This also updates the Events Library submodule to v0.4.1, adding in configuration profiles.

File: platform/ios/src/MGLMapboxEvents.m [hunk 1]
```diff
@@ -8,6 +8,8 @@
 static NSString * const MGLMapboxMetricsDebugLoggingEnabled = @"MGLMapboxMetricsDebugLoggingEnabled";
 static NSString * const MGLTelemetryAccessToken = @"MGLTelemetryAccessToken";
 static NSString * const MGLTelemetryBaseURL = @"MGLTelemetryBaseURL";
+static NSString * const kMMEEventsProfile = @"MMEEventsProfile";
```
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: Let’s make the naming for these consistent with the other constants (e.g., remove the `k` Hungarian notation, use a consistent prefix, etc.).

File: platform/ios/src/MGLMapboxEvents.m [hunk 2]
```diff
@@ -8,6 +8,8 @@
...
+static NSString * const kConfigHibernationRadiusVariableKey = @"VariableGeofence";
```
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: The naming of `kConfigHibernationRadiusVariableKey` is a little unclear to me — this is the _value_ that the `kMMEEventsProfile` _key_ is checked for, so this shouldn’t include `Key` in its name.

File: platform/ios/src/MGLMapboxEvents.m [hunk 3]
```diff
@@ -122,21 +124,31 @@ - (void)updateDisablingConfigurationValuesWithNotification:(NSNotification *)not
 }
 
 + (void)setupWithAccessToken:(NSString *)accessToken {
-    NSString *semanticVersion = [NSBundle mgl_frameworkInfoDictionary][@"MGLSemanticVersionString"];
-    NSString *shortVersion = [NSBundle mgl_frameworkInfoDictionary][@"CFBundleShortVersionString"];
-    NSString *sdkVersion = semanticVersion ?: shortVersion;
+    int delayTime = 0;
```
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: If possible, use the more specific `dispatch_time_t` type here.

Comment (friedbunny) [reply platform/ios/src/MGLMapboxEvents.m]: Oops, I thought I deleted this comment — it’s incorrect. The more specific type would be `uint_64`, but using `int` probably isn’t the end of the world.

**Pull request merged**